solutions/usaco-1011.mdx

```diff
@@ -95,4 +95,58 @@ public class triangles{
 
 </JavaSection>
 
+<CPPSection>
+
+```CPP
+
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	freopen("triangles.in", "r", stdin);
+	freopen("triangles.out", "w", stdout);
+	int n; cin >> n;
+	int x[n];
+	int y[n];
+	int ans = 0;
+	for (int i = 0; i < n; i++) {
+		cin >> x[i] >> y[i];
+	}
+	for (int i = 0; i < n; i++) {
+		for (int j = 0; j < n; j++) {
+			for (int k = 0; k < n; k++) {
+				// Check to see if both points are parallel to the X and Y axis
+				if (y[i] ==y [j] && x[i] == x[k]) {
+					/** 
+					 * Since we need to output two times the
+					 * area and the formula for the area
+					 * of a triangle is 1/2 * base * height, 
+					 * we can omit the 1/2 because it anyways
+					 * gets multiplied by 2, which cancels it out. 
+					 * Finding the base is the difference
+					 * between both x points and the height is 
+					 * the difference between both y points;
+					 * it is a right triangle because both 
+					 * sides are parallel to the x and y axis,
+					 * so no advanced math is required to find 
+					 * the height.
+					 */
+					int area = (x[j] - x[i]) * (y[k] - y[i]);
+					if (area < 0) {
+						area *= -1;
```

Comment:
```suggestion
						area = -area;
```

Reply:
resolving this one because I feel like there is a better way (addressed in my changes [here](https://github.com/cpinitiative/usaco-guide/pull/2291#discussion_r739871558)